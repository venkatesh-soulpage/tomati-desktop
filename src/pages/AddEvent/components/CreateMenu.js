import React from "react";
import { Form, Button, Card } from "react-bootstrap";
import Papa from "papaparse";
import _ from "lodash";

const CreateMenu = ({
  values,
  handleChange,
  handleCreateOutlet,
  setValues,
}) => {
  const uploadFile = (data) => {
    const { data: csv_data } = data;
    setValues({ ...values, menu: _.reject(csv_data, { name: "" }) });
  };
  return (
    <div>
      <Form.Group>
        <Form.Control
          as="select"
          placeholder="Location"
          value={values.location}
          onChange={handleChange("location_id")}
          required
        >
          <option>Select Location</option>
          <option value={7}>Nigeria</option>
          <option disabled>Angola (Coming Soon)</option>
          <option disabled>Brazil (Coming Soon)</option>
          <option disabled>Colombia (Coming Soon)</option>
          <option disabled>France (Coming Soon)</option>
          <option disabled>Ghana (Coming Soon)</option>
          <option disabled>Kenya (Coming Soon)</option>
          <option disabled>Poland (Coming Soon)</option>
          <option disabled>South Africa (Coming Soon)</option>
          <option disabled>Spain (Coming Soon)</option>
          <option disabled>United Arab Emirates (Coming Soon)</option>
          <option disabled>United Kingdom (Coming Soon)</option>
        </Form.Control>
      </Form.Group>
      <Form.Group>
        <Form.File
          id="coverImage"
          accept=".csv"
          label="Custom file input"
          custom
          className="d-none"
          onChange={(e) => {
            Papa.parse(e.target.files[0], {
              complete: uploadFile,
              header: true,
              transformHeader: (header) =>
                header.toLowerCase().replace(/\W/g, "_"),
            });
          }}
        />
        <Card className="p-4">
          <label for="coverImage">
            <h6>Upload Menu</h6>
          </label>
        </Card>
      </Form.Group>
      <Form.Group className="d-flex justify-content-end">
        <Button
          form="register-form"
          className="btn btn-primary mt-3"
          style={{ borderRadius: "20px" }}
          onClick={handleCreateOutlet}
        >
          Continue
        </Button>
      </Form.Group>
    </div>
  );
};

export default CreateMenu;
