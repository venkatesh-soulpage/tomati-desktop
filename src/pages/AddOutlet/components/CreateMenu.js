import React from "react";
import { Form, Button, Card } from "react-bootstrap";
import Papa from "papaparse";
import _ from "lodash";
import Back from "assets/img/Back.svg";
import UploadCover from "assets/img/UploadCover.svg";

const CreateMenu = ({
  values,
  handleChange,
  handleCreateOutlet,
  setValues,
  handleStepPrev,
}) => {
  const uploadFile = (data) => {
    const { data: csv_data } = data;
    setValues({ ...values, menu: _.reject(csv_data, { name: "" }) });
  };
  return (
    <div>
      <Form.Group>
        <Form.Control
          as="select"
          placeholder="Location"
          value={values.location}
          onChange={handleChange("location_id")}
          required
        >
          <option>Select Location</option>
          <option value={7}>Nigeria</option>
          <option disabled>Angola (Coming Soon)</option>
          <option disabled>Brazil (Coming Soon)</option>
          <option disabled>Colombia (Coming Soon)</option>
          <option disabled>France (Coming Soon)</option>
          <option disabled>Ghana (Coming Soon)</option>
          <option disabled>Kenya (Coming Soon)</option>
          <option disabled>Poland (Coming Soon)</option>
          <option disabled>South Africa (Coming Soon)</option>
          <option disabled>Spain (Coming Soon)</option>
          <option disabled>United Arab Emirates (Coming Soon)</option>
          <option disabled>United Kingdom (Coming Soon)</option>
        </Form.Control>
      </Form.Group>
      <Form.Group>
        <Form.File
          id="menu"
          accept=".csv"
          label="Custom file input"
          custom
          className="d-none"
          onChange={(e) => {
            Papa.parse(e.target.files[0], {
              complete: uploadFile,
              header: true,
              transformHeader: (header) =>
                header.toLowerCase().replace(/\W/g, "_"),
            });
          }}
        />
        <Card
          style={{ border: " 1px dashed", cursor: "pointer" }}
          className="p-2 d-flex pt-4"
        >
          <label for="menu" style={{ cursor: "pointer" }}>
            <h6>
              <img src={UploadCover} alt="icon" className="mx-4" />
              {values.menu ? (
                <span>{values.menu[0].name}</span>
              ) : (
                <span>Upload Menu</span>
              )}
            </h6>
          </label>
        </Card>
        <h6 className="mt-2" style={{ color: "#989CA4" }}>
          Only CSV Files
        </h6>
      </Form.Group>
      <Form.Group className="d-flex justify-content-between">
        <img
          className="mt-3"
          style={{ height: "54px", cursor: "pointer" }}
          src={Back}
          alt="icon"
          onClick={() => {
            handleStepPrev("step", 1);
          }}
        />
        <Button
          form="register-form"
          className="btn btn-primary mt-3 rounded-pill px-4"
          style={{ borderRadius: "20px" }}
          onClick={handleCreateOutlet}
        >
          Continue
        </Button>
      </Form.Group>
    </div>
  );
};

export default CreateMenu;
