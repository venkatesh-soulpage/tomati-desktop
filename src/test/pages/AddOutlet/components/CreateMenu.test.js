import React from "react";
import { render } from "@testing-library/react";
import CreateMenu from "pages/AddOutlet/components/CreateMenu";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("CreateMenu page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <CreateMenu />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
