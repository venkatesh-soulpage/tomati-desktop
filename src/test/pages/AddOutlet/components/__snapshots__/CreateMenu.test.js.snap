// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateMenu page matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="form-group"
    >
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="custom-dropdown text-left d-flex justify-content-between align-items-center btn dropdown-toggle btn btn-primary"
          type="button"
        >
          Select Locations (required)
        </button>
      </div>
    </div>
    <div
      class="form-group"
    >
      <div
        class="d-none custom custom-file"
      >
        <input
          accept=".csv"
          class="custom-file-input"
          id="menu"
          type="file"
        />
        <label
          class="custom-file-label"
          for="menu"
        >
          Custom file input
        </label>
      </div>
      <div
        class="p-2 d-flex pt-4 b1-dash cr-p card"
      >
        <label
          class="cr-p"
          for="menu"
        >
          <h6>
            <img
              alt="icon"
              class="mx-4"
              src="UploadCover.svg"
            />
            <span>
              Upload Menu
            </span>
          </h6>
        </label>
      </div>
      <h6
        class="mt-2 grey-color"
      >
        Only CSV Files
      </h6>
      <h6
        class="mt-2 grey-color font-italic"
      >
        watch the CSV tutorial here
      </h6>
      <h6
        class="mt-2 grey-color font-italic"
      >
        download CSV template here
      </h6>
    </div>
    <div
      class="d-flex justify-content-between form-group"
    >
      <img
        alt="icon"
        class="mt-3 ht-54 cr-p"
        src="Back.svg"
      />
      <button
        class="btn btn-primary mt-3 rounded-pill px-4 btn btn-primary"
        form="email-form"
        type="submit"
      >
        Continue
      </button>
    </div>
  </div>
</DocumentFragment>
`;
